### PR TITLE
Add version config

### LINKED ISSUE #3155: [FEA] CUDF_VERSION in cudf.h ?
**Is your feature request related to a problem? Please describe.**
Changes to cudf functionality may break existing code that relies on cudf. The lack of versioning makes it difficult to have a code base that could support multiple cudf versions. 

**Describe the solution you'd like**
Seems like a CUDF_VERSION, perhaps along with CUDF_VERSION_{MAJOR|MINOR} definitions would be useful. Would also be useful for cudf-io writers for putting cudf version in parquet/orc files.

### PR DESCRIPTION
**Pull request by vyasr**
## Description

Resolves #3155.

## Checklist
- [x] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cudf/blob/HEAD/CONTRIBUTING.md).
- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.

### CODE REVIEW

**Review by bdice — APPROVED** [review]

File: cpp/CMakeLists.txt [hunk 1]
```diff
@@ -36,6 +36,8 @@ if(CMAKE_CUDA_COMPILER_ID STREQUAL "NVIDIA" AND CMAKE_CUDA_COMPILER_VERSION VERS
   )
 endif()
 
+rapids_cmake_write_version_file(include/cudf/version_config.hpp)
```
Comment (robertmaynard) [inline cpp/CMakeLists.txt:39]: We need an install rule for this file

**Review by robertmaynard — CHANGES_REQUESTED** [review]

File: cpp/CMakeLists.txt (hunk 1 above)
Comment (vyasr) [reply cpp/CMakeLists.txt:39]: Oh whoops good call. I somehow though that was automatically handled by the rapids-cmake function.

**Comment by vyasr** [comment]
Cool @robertmaynard I think this is good to go when you approve.

**Review by robertmaynard — APPROVED** [review]

**Pull request merged**